main/src/main/java/org/mobilitydata/gtfsvalidator/validator/FareTransferRuleTransferCountValidator.java

```diff
@@ -0,0 +1,83 @@
+package org.mobilitydata.gtfsvalidator.validator;
+
+import java.util.Objects;
+import org.mobilitydata.gtfsvalidator.annotation.GtfsValidator;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.notice.SeverityLevel;
+import org.mobilitydata.gtfsvalidator.notice.ValidationNotice;
+import org.mobilitydata.gtfsvalidator.table.GtfsFareTransferRule;
+
+@GtfsValidator
+public class FareTransferRuleTransferCountValidator
+    extends SingleEntityValidator<GtfsFareTransferRule> {
+
+  @Override
+  public void validate(GtfsFareTransferRule rule, NoticeContainer noticeContainer) {
+    if (rule.hasFromLegGroupId()
+        && rule.hasToLegGroupId()
+        && Objects.equals(rule.fromLegGroupId(), rule.toLegGroupId())) {
+      if (rule.hasTransferCount()) {
+        if (rule.transferCount() < -1 || rule.transferCount() == 0) {
+          noticeContainer.addValidationNotice(
+              new InvalidFareTransferRuleTransferCountNotice(
+                  rule.csvRowNumber(), rule.transferCount()));
+        }
+      } else {
+        noticeContainer.addValidationNotice(
+            new FareTransferRuleMissingTransferCountNotice(rule.csvRowNumber()));
+      }
+    } else {
+      if (rule.hasTransferCount()) {
+        noticeContainer.addValidationNotice(
+            new FareTransferRuleWithForbiddenTransferCountNotice(rule.csvRowNumber()));
+      }
+    }
+  }
+
+  /**
+   * A row from GTFS file `fare_transfer_rules.txt` has a defined `transfer_count` with an invalid
+   * value.
+   *
+   * <p>Severity: {@code SeverityLevel.ERROR}
+   */
+  static class InvalidFareTransferRuleTransferCountNotice extends ValidationNotice {
+    private final long csvRowNumber;
+    private final int transfer_count;
```

Comment:
Could we update this variable to the camel case format for consistency?

Reply:
Done.